cms/core/templatetags/util_tags.py

```diff
@@ -37,3 +38,17 @@ def social_image(page: "Page", site: "Site") -> Optional["CustomImage"]:
 def include_django(context: jinja2.runtime.Context, template_name: str) -> "SafeString":
     """Allow importing a pre-rendered Django template into jinja2."""
     return render_to_string(template_name, context=dict(context), request=context.get("request", None))
+
+
+# Breadcrumbs
+@register.filter(name="breadcrumbs")
+def breadcrumbs(page: "Page") -> list[dict[str, str]]:
```

Comment:
suggestion

```suggestion
@jinja2.pass_context
def breadcrumbs(context: jinja2.runtime.Context, page: "Page") -> list[dict[str, str]]:
```

a la https://github.com/ONSdigital/dis-wagtail/pull/85/files#diff-bfd42dbff73bbdea18630c438176ac08db2c376e9a8c3205a430e5f15caa3264R75-R76

then you can do `.get_url(request=context.get("request"))`.


`@register.filter(name="breadcrumbs")` is for the Django template library, which we only use for the Wagtail admin

Reply:
Could you also update the breadcrumbs template to remove https://github.com/ONSdigital/dis-wagtail/blob/main/cms/jinja2/templates/components/navigation/breadcrumbs.html#L4-L14 and change https://github.com/ONSdigital/dis-wagtail/blob/main/cms/jinja2/templates/components/navigation/breadcrumbs.html#L19 to `page|breadcrumbs` ?

Reply:
I think it would make sense to use the `navigation` app for the breadcrumbs like the ones in @sanjeevz3009's merge request so will update.

Reply:
> it seems to be being used for the social_image and social_text filters currently which are used on the front-end not in the admin.

that was part of kit and it makes sense to clean them

Reply:
Happy to leave them if that makes sense - did you mean 'keep them' not 'clean them'?

Reply:
I will leave the breadcrumbs template for now and see what becomes of this pull request - it may be that we can get rid of it entirely once we're using the new heroes more consistently.